nilearn/datasets/func.py

```diff
@@ -1411,3 +1412,98 @@ def fetch_megatrawls_netmats(dimensionality=100, timeseries='eigen_regression',
         matrices=matrices,
         correlation_matrices=correlation_matrices,
         description=description)
+
+
+def fetch_cobre(n_subjects=10, data_dir=None, url=None, verbose=1):
+    """Fetch COBRE datasets preprocessed using NIAK 0.12.4 pipeline.
+
+    Downloads and returns preprocessed resting state fMRI datasets and
+    phenotypic information such as demographic, clinical variables,
+    measure of frame displacement FD (an average FD for all the time
+    frames left after censoring).
+
+    For each subject, this function also returns .mat files which contains
+    all the covariates that have been regressed out of the functional data.
+    The covariates such as motion parameters, mean CSF signal, etc. It also
+    contains a list of time frames that have been removed from the time series
+    by censoring for high motion.
+
+    NOTE: The number of time samples vary, as some samples have been removed
+    if tagged with excessive motion. This means that data is already time
+    filtered. See output variable 'description' for more details.
+
+    .. versionadded 0.2.3
+
+    Parameters
+    ----------
+    n_subjects: int, optional
+        The number of subjects to load from maximum of 146 subjects.
+        By default, 10 subjects will be loaded. If n_subjects=None,
+        all subjects will be loaded.
+
+    data_dir: str, optional
+        Path to the data directory. Used to force data storage in a
+        specified location. Default: None
+
+    url: str, optional
+        Override download url. Used for test only (or if you setup a
+        mirror of the data). Default: None
+
+    verbose: int, optional
+       Verbosity level (0 means no message).
+
+    Returns
+    -------
+    data: Bunch
+        Dictionary-like object, the attributes are:
+
+        - 'func': string list
+            Paths to Nifti images.
+        - 'mat_files': string list
+            Paths to .mat files of each subject.
+        - 'phenotypic': ndarray
+            Contains data of clinical variables, sex, age, FD.
+        - 'description': data description of the release and references.
+    """
+    if url is None:
+        url = "https://ndownloader.figshare.com/articles/1160600/versions/15"
```

Comment:
Well, Gaël said in the issue that a 1 hour download was too much. Plus it does not make sense to return 10 subjects by default if you force downloading the whole dataset. But nevermind, let's move on.